src/payment/strategies/cybersource/cybersource-payment-strategy.ts

```diff
@@ -0,0 +1,140 @@
+import { some } from 'lodash';
+
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import {
+    MissingDataError,
+    MissingDataErrorType,
+    RequestError
+} from '../../../common/error/errors';
+import { OrderActionCreator, OrderPaymentRequestBody, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import Payment, {CreditCardInstrument, PaymentInstrument, ThreeDSecureToken} from '../../payment';
+import PaymentActionCreator from '../../payment-action-creator';
+import PaymentMethod from '../../payment-method';
+import PaymentMethodActionCreator from '../../payment-method-action-creator';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import { CardinalClient, CardinalOrderData } from './index';
+
+export default class CyberSourcePaymentStrategy implements PaymentStrategy {
+    private _paymentMethod?: PaymentMethod;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _cardinalClient: CardinalClient
+    ) {}
+
+    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options;
+
+        return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId)).then( state => {
+            this._paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+
+            if (!this._paymentMethod || !this._paymentMethod.config || this._paymentMethod.config.testMode === undefined) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
+
+            return this._cardinalClient.initialize(this._paymentMethod.config.testMode).then(() => this._store.getState());
+        });
+    }
+
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+
+        if (!payment || !payment.paymentData) {
+            throw new MissingDataError(MissingDataErrorType.MissingPayment);
+        }
+
+        if (!this._paymentMethod || !this._paymentMethod.config) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return !this._paymentMethod.config.is3dsEnabled || !payment.paymentData.ccNumber ? this._placeOrder(order, payment, payment.paymentData, options) :
```

Comment:
it will happen when users use vaulted credit cards, however we changed this code to `this._paymentMethod.config.isVaultingEnabled`, we think this is the correct way to validate it